splink/comparison_level_creator.py

```diff
@@ -0,0 +1,98 @@
+from abc import ABC, abstractmethod
+from typing import final
+
+from .comparison_level import ComparisonLevel
+from .input_column import InputColumn
+
+
+class ComparisonLevelCreator(ABC):
+    def __init__(self, col_name: str = None):
+        """_summary_
+
+        Args:
+            col_name (str): _description_
+            dialect (str, optional): _description_. Defaults to None.
+            m_probability (_type_, optional): _description_. Defaults to None.
+        """
+        self.col_name = col_name
+        # TODO: do we need to set these to None, or let ComparisonLevel handle missings?
+        self.m_probability = None
+        self.u_probability = None
+        self.tf_adjustment_column = None
+        self.tf_adjustment_weight = None
+        self.tf_minimum_u_value = None
+        self.is_null_level = None
+
+    @abstractmethod
+    def create_sql(self, sql_dialect: str):
+        pass
+
+    @abstractmethod
+    def create_label_for_charts(self):
+        pass
+
+    @final
+    def get_comparison_level(self, sql_dialect: str):
+        return ComparisonLevel(self.create_level_dict(sql_dialect))
```

Comment:
is there an argument for turning sql_dialect into a SplinkDialect here? so that e.g. the 
```
def create_sql(self, dialect:SplinkDialect):
        col = self.input_column(dialect) #have to modify self.input_column fn slightly
        lev_fn = dialect.levenshtein_function_name
        return f"{lev_fn}({col.name_l()}, {col.name_r()}) <= {self.distance_threshold}"
        ```

Reply:
Yeah I think this is simpler than the contextmanager stuff. Also feels quite natural in that whoever is adding these levels will usually have to be making changes to `SplinkDialect` and it's subclasses anyhow, so it's natural to then be working with these directly when making the sql rather than needing to go via a lookup